lib/backend/firestore/firestorebk_test.go

```diff
@@ -270,7 +270,10 @@ func TestDeleteDocuments(t *testing.T) {
 
 			lis, err := net.Listen("tcp", "localhost:0")
 			require.NoError(t, err)
-			go func() { require.NoError(t, srv.Serve(lis)) }()
+
+			errCh := make(chan error)
+			go func() { errCh <- srv.Serve(lis) }()
+			t.Cleanup(func() { require.NoError(t, <-errCh) })
```

Comment:
Should we do a single cleanup that does Stop and then receives from errCh? It's a bit more resistant to eventual changes.

Reply:
We should be able do that so long as we buffer the channel. Otherwise the stacked behavior of cleanups would create a deadlock. Let me give it a try.